frontend/packages/dev-console/src/utils/add-resources-menu-utils.ts

```diff
@@ -0,0 +1,74 @@
+import * as _ from 'lodash';
+import { K8sResourceKind, modelFor, referenceFor } from '@console/internal/module/k8s';
+import { KebabOption, asAccessReview } from '@console/internal/components/utils';
+import { ImportOptions } from '../components/import/import-types';
+
+const PART_OF = 'app.kubernetes.io/part-of';
+
+export const getAddPageUrl = (
+  obj: K8sResourceKind,
+  type: string,
+  hasApplication: boolean,
+): string => {
+  let pageUrl = '';
+  const params = new URLSearchParams();
+  const {
+    metadata: { namespace: ns },
+  } = obj;
+  switch (type) {
+    case ImportOptions.GIT:
+      pageUrl = `/import/ns/${ns}`;
+      params.append('importType', 'git');
+      break;
+    case ImportOptions.CONTAINER:
+      pageUrl = `/deploy-image/ns/${ns}`;
+      params.append('preselected-ns', ns);
+      break;
+    case ImportOptions.CATALOG:
+      pageUrl = '/catalog';
+      break;
+    case ImportOptions.DOCKERFILE:
+      pageUrl = `/import/ns/${ns}`;
+      params.append('importType', 'docker');
+      break;
+    case ImportOptions.DATABASE:
+      pageUrl = `/catalog/ns/${ns}`;
+      params.append('category', 'databases');
+      break;
+    default:
+      pageUrl = `add/ns/${ns}`;
```

Comment:
Hmm, I probably should handle this in my PR... Since if they land there now, they'll likely lose all their params on the first navigate.